packages/js/src/analysis/plugins/shortcode-plugin.js

```diff
@@ -0,0 +1,324 @@
+		this.closingTagRegex = new RegExp( "\\[\\/" + keywordRegexString + "\\]", "g" );
+		this.nonCaptureRegex = new RegExp( "\\[" + keywordRegexString + "[^\\]]*?\\]", "g" );
+
+		/**
+		 * The array of parsedShortcode objects.
+		 *
+		 * @type {Object[]}
+		 */
+		this.parsedShortcodes = [];
+
+		this.loadShortcodes( this.declareReady.bind( this ) );
+	}
+
+	/* YOAST SEO CLIENT */
+
+	/**
+	 * Declares ready with YoastSEO.
+	 *
+	 * @returns {void}
+	 */
+	declareReady() {
+		this._pluginReady( "YoastShortcodePlugin" );
+		this.registerModifications();
+	}
+
+	/**
+	 * Declares reloaded with YoastSEO.
+	 *
+	 * @returns {void}
+	 */
+	declareReloaded() {
+		this._pluginReloaded( "YoastShortcodePlugin" );
+	}
+
+	/**
+	 * Registers the modifications for the content in which we want to replace shortcodes.
+	 *
+	 * @returns {void}
+	 */
+	registerModifications() {
+		this._registerModification( "content", this.replaceShortcodes.bind( this ), "YoastShortcodePlugin" );
+	}
+
+
+	/**
+	 * Removes all unknown shortcodes. Not all plugins properly registered their shortcodes in the WordPress backend.
+	 * Since we cannot use the data from these shortcodes they must be removed.
+	 *
+	 * @param {String} data The text to remove unknown shortcodes.
+	 * @returns {String} The text with removed unknown shortcodes.
+	 */
+	removeUnknownShortCodes( data ) {
+		data = data.replace( shortcodeStartRegex, "" );
+		data = data.replace( shortcodeEndRegex, "" );
+
+		return data;
+	}
+
+	/**
+	 * Replaces the unparsed shortcodes with the parsed ones.
+	 * The callback used to replace the shortcodes.
+	 *
+	 * @param {String} data The text to replace the shortcodes in.
+	 *
+	 * @returns {String} The text with replaced shortcodes.
+	 */
+	replaceShortcodes( data ) {
+		const parsedShortcodes = this.parsedShortcodes;
+
+		if ( typeof data === "string" && parsedShortcodes.length > 0 ) {
+			parsedShortcodes.forEach( ( { shortcode, output } ) => {
+				data = data.replace( shortcode, output );
+			} );
+		}
+
+		data = this.removeUnknownShortCodes( data );
+
+		return data;
+	}
+
+	/* DATA SOURCING */
+
+	/**
+	 * Gets data from input fields and store them in an analyzerData object. This object will be used to fill
+	 * the analyzer and the snippet preview.
+	 *
+	 * @param {function} callback To declare either ready or reloaded after parsing.
+	 *
+	 * @returns {void}
+	 */
+	loadShortcodes( callback ) {
+		const unparsedShortcodes = this.getUnparsedShortcodes( this.getShortcodes( this.getContentTinyMCE() ) );
+		if ( unparsedShortcodes.length > 0 ) {
+			this.parseShortcodes( unparsedShortcodes, callback );
+		} else {
+			return callback();
+		}
+	}
+
+	/**
+	 * Binds elements to be able to reload the dataset if shortcodes get added.
+	 *
+	 * @returns {void}
+	 */
+	bindElementEvents() {
+		const contentElement = document.getElementById( "content" ) || false;
+		const callback =  _.debounce(	this.loadShortcodes.bind( this, this.declareReloaded.bind( this ) ), 500 );
```

Comment:
Some unnecessary spacing here. Also, I think it's best to properly import from `lodash` here instead of relying on a global. For the other globals I think we can be a little bit more forgiving.